woocommerce/woocommerce
# [Experimental] Product filters: Add insertion feature

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds the ability for filters to be added back within the parent wrapper container.

NOTE:

- I also took this opportunity to update the API version to 3 to align with other filters.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Test by developers only**

Prereq:
- Ensure you have the WC Beta Tester Plugin installed and go to Tools > WCA Test Helper > Features. Toggle on the experimental-blocks option.

1. Go to the editor and navigate to Product Catalog template.
2. Add the Product Filters (Experimental) block to the editor.
3. Delete any of the filter (Active,Status,Price,Attributes,Rating).
4. Within the Product Filters Wrapper, use the inserter + button and try to add back the filters.
5. Ensure you can add each one of the filters back.
6. Try adding more than one of the same filter. You should see a warning stating one has already be added. Attributes filter is the exception. We will allow multiples of this filter.
7. Ensure you still can't insert any of the individual filters outside of the wrapper block.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Add ability to add individual filters back
</details>
